Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -444,27 +528,41 @@ def preprocess_url(url)
     # the version information. Returns nil if a latest version couldn't be found.
     sig {
       params(
-        formula_or_cask: T.any(Formula, Cask::Cask),
-        json:            T::Boolean,
-        full_name:       T::Boolean,
-        verbose:         T::Boolean,
-        debug:           T::Boolean,
+        formula_or_cask:           T.any(Formula, Cask::Cask),
+        livecheck_formula_or_cask: T.nilable(T.any(Formula, Cask::Cask)),
+        livecheck_references:      T::Array[T.any(Formula, Cask::Cask)],
+        json:                      T::Boolean,
+        full_name:                 T::Boolean,
+        verbose:                   T::Boolean,
+        debug:                     T::Boolean,
       ).returns(T.nilable(Hash))
     }
-    def latest_version(formula_or_cask, json: false, full_name: false, verbose: false, debug: false)
+    def latest_version(
+      formula_or_cask,
+      livecheck_formula_or_cask: nil,
```

Comment:
I agree that `livecheck_formula_or_cask` isn't as clear as it could be about what it actually is. I think `reference_formula_or_cask` would probably be a good fit without being too verbose.